src/utils/addCustomOption.js

```diff
@@ -1,15 +1,28 @@
 import {uniqueId} from 'lodash';
 import {getOptionLabel, getStringLabelKey} from './index';
 
-function addCustomOption(results, text, labelKey) {
-  const exactMatchFound = results.some((o) => (
-    getOptionLabel(o, labelKey) === text
-  ));
-
-  if (!text.trim() || exactMatchFound) {
+function addCustomOption(results, text, labelKey, props) {
```

Comment:
Given that `text` and `labelKey` are part of `props`, I think it makes sense to just collapse the signature: 

```jsx
addCustomOption(results, props) { ... }
```